includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -513,6 +513,35 @@ protected function prepare_payment_information( $order ) {
 		return $payment_information;
 	}
 
+	/**
+	 * Upserts customer from user, name, and email.
+	 *
+	 * @param WP_User $user The user associated to customer.
+	 * @param string  $name The customer's name.
+	 * @param string  $email The customer's email.
+	 *
+	 * @return string|null Customer id.
+	 */
+	protected function upsert_customer( $user, $name, $email ) {
+		if ( ! $user ) {
+			return;
+		}
```

Comment:
Comparing to the original code, this if-statement looks like dead code. Any specific reason why we adding it?

Reply:
Ah you're right. The only possible source of $user is from this statement
```php
$order->get_user() ?? wp_get_current_user()
```
`$order->get_user()` alone can return false but `wp_get_current_user()` always return `WP_User`.

I'm removing this check.